packages/open-payments/src/types.ts

```diff
@@ -1,9 +1,44 @@
-import { components, paths as Paths } from './generated/types'
+import {
+  components as RSComponents,
+  paths as RSPaths
+} from './generated/resource-server-types'
+import {
+  components as ASComponents,
+  paths as ASPaths,
+  operations as ASOperations
+} from './generated/auth-server-types'
 
-export const getPath = <P extends keyof Paths>(path: P): string =>
+export const getRSPath = <P extends keyof RSPaths>(path: P): string =>
   path as string
-
 export type IncomingPayment =
-  components['schemas']['incoming-payment-with-connection']
-export type ILPStreamConnection = components['schemas']['ilp-stream-connection']
-export type PaymentPointer = components['schemas']['payment-pointer']
+  RSComponents['schemas']['incoming-payment-with-connection']
+export type ILPStreamConnection =
+  RSComponents['schemas']['ilp-stream-connection']
+export type PaymentPointer = RSComponents['schemas']['payment-pointer']
+
+export const getASPath = <P extends keyof ASPaths>(path: P): string =>
+  path as string
+export type NonInteractiveGrantRequest = {
+  access_token: ASOperations['post-request']['requestBody']['content']['application/json']['access_token']
+  client: ASOperations['post-request']['requestBody']['content']['application/json']['client']
+}
+export type NonInteractiveGrant = {
+  access_token: ASComponents['schemas']['access_token']
+  continue: ASComponents['schemas']['continue']
+}
+export type GrantRequest = {
+  access_token: ASOperations['post-request']['requestBody']['content']['application/json']['access_token']
+  client: ASOperations['post-request']['requestBody']['content']['application/json']['client']
+  interact?: ASOperations['post-request']['requestBody']['content']['application/json']['interact']
+}
+export type InteractiveGrant = {
+  interact: ASComponents['schemas']['interact-response']
+  continue: ASComponents['schemas']['continue']
+}
+export const isInteractiveGrant = (
+  grant: InteractiveGrant | NonInteractiveGrant
+): grant is InteractiveGrant => !!(grant as InteractiveGrant).interact
+
+export const isNonInteractiveGrant = (
+  grant: InteractiveGrant | NonInteractiveGrant
+): grant is NonInteractiveGrant => !!(grant as NonInteractiveGrant).access_token
```

Comment:
could live in `src/client/grants.ts` or `src/type-guard.ts` but decided to leave them here